From: Takashi Iwai <tiwai@suse.de>
Subject: Workaround kABI breakage by __must_check drop of strscpy()
Patch-mainline: Never, kABI fix
References: bsc#1051510

kABI checker is so captious and complains when __must_check annotation
is dropped.  Show him the original declaration.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/linux/string.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/linux/string.h
+++ b/include/linux/string.h
@@ -27,8 +27,12 @@ extern char * strncpy(char *,const char
 size_t strlcpy(char *, const char *, size_t);
 #endif
 #ifndef __HAVE_ARCH_STRSCPY
+#ifdef __GENKSYMS__
+ssize_t __must_check strscpy(char *, const char *, size_t);
+#else
 ssize_t strscpy(char *, const char *, size_t);
 #endif
+#endif
 #ifndef __HAVE_ARCH_STRCAT
 extern char * strcat(char *, const char *);
 #endif
